src/main/kotlin/uk/gov/gdx/datashare/config/R2dbcConfig.kt

```diff
@@ -0,0 +1,72 @@
+package uk.gov.gdx.datashare.config
+
+import io.r2dbc.pool.ConnectionPool
+import io.r2dbc.pool.ConnectionPoolConfiguration
+import io.r2dbc.spi.*
+import org.reactivestreams.Publisher
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Primary
+import org.springframework.data.r2dbc.config.AbstractR2dbcConfiguration
+import reactor.core.publisher.Mono
+import uk.gov.gdx.datashare.service.DbPasswordService
+
+@Configuration
+class R2dbcConfig(
+  @Autowired val dbPasswordService: DbPasswordService,
```

Comment:
does this need the autowired attribute, or is that just for clarity?

Reply:
It does not 🙂 That's kruft left over from me trying something else